src/test/scala/edu/ie3/simona/model/participant/BMModelSpec.scala

```diff
@@ -0,0 +1,338 @@
+/*
+ * © 2020. TU Dortmund University,
+ * Institute of Energy Systems, Energy Efficiency and Energy Economics,
+ * Research group Distribution grid planning and operation
+ */
+
+package edu.ie3.simona.model.participant
+
+import edu.ie3.datamodel.models.input.system.`type`.BmTypeInput
+import edu.ie3.datamodel.models.input.system.characteristic.CosPhiFixed
+import edu.ie3.simona.model.participant.ModelState.ConstantState
+import edu.ie3.simona.model.participant.control.QControl
+import edu.ie3.simona.test.common.UnitSpec
+import edu.ie3.util.scala.OperationInterval
+import edu.ie3.util.scala.quantities.EuroPerKilowatthour
+import squants.energy.{Kilowatts, Megawatts}
+import squants.market.EUR
+import squants.thermal.Celsius
+import squants.{Power, Temperature}
+import tech.units.indriya.quantity.Quantities
+
+import java.time.ZonedDateTime
+import java.util.UUID
+
+/** Test class that tries to cover all special cases of the current
+  * implementation of the {@link BMModel}
+  *
+  * Test results have been calculated on paper using equations from wiki:
+  * https://wiki.ie3.e-technik.tu-dortmund.de/!simona/model:bm_model
+  */
+
+class BMModelSpec extends UnitSpec {
+
+  implicit val powerTolerance: Power = Megawatts(1e-4)
+  implicit val power2Tolerance: Power = Kilowatts(1e-4)
+  implicit val usageTolerance: Double = 1e-12
+
+  val bmType: BmTypeInput = new BmTypeInput(
+    UUID.fromString("bc06e089-03cd-481e-9e28-228266a148a4"),
+    "BM Model Test Type 1",
+    Quantities.getQuantity(0, edu.ie3.util.quantities.PowerSystemUnits.EURO),
+    Quantities.getQuantity(
+      0.05d,
+      edu.ie3.util.quantities.PowerSystemUnits.EURO_PER_KILOWATTHOUR,
+    ),
+    Quantities.getQuantity(
+      5,
+      edu.ie3.util.quantities.PowerSystemUnits.PERCENT_PER_HOUR,
+    ),
+    Quantities.getQuantity(
+      190,
+      edu.ie3.util.quantities.PowerSystemUnits.KILOVOLTAMPERE,
+    ),
+    1d,
+    Quantities.getQuantity(100d, tech.units.indriya.unit.Units.PERCENT),
+  )
```

Comment:
this BmTypeInput isn't realy used much. Please delete this and insert the values (1d for cosPhiRated and EUR(0.05)) within the methods directly.